testplan/runnable/interactive/base.py

```diff
@@ -279,6 +283,78 @@ def test(self, test_uid, runner_uid=None):
         item = runner.added_item(test_uid)
         return item
 
+    def start_test_resources(
+        self, test_uid, runner_uid=None, await_results=True
+    ):
+        """
+        Start all test resources.
+
+        :param test_uid: UID of test to start resources for
+        :param runner_uid: UID of test runner that owns the test, or None to
+            specify the default local runner.
+        :param await_results: Whether to block until the test resources have
+            all started, defaults to True.
+        :return: If await_results is True, returns a list of the return values
+            of each resource start operation, otherwise returns a list of async
+            result objects.
+        """
+        test = self.test(test_uid, runner_uid=runner_uid)
+        test_irunner = test.cfg.interactive_runner(test)
+
+        results = self._run_all_test_operations(
+            test_irunner.start_resources(), await_results
+        )
+
+        if await_results:
+            results.append(
+                self._set_env_status(test_uid, entity.ResourceStatus.STARTED)
```

Comment:
How do we know the env is indeed started?

Reply:
When `await_results=True` the environment is started synchronously, so by the time we get to here the start actions must have completed successfully (otherwise an exception would be raised).

In the async case, we should add a check that the environment was started without error.